dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/User.java

```diff
@@ -287,16 +289,20 @@ public String getUserRoleNames()
      */
     public Set<String> getAllAuthorities()
     {
+        if ( userRoles.isEmpty() )
+        {
+            return emptySet();
+        }
+        if ( userRoles.size() == 1 )
```

Comment:
Just curious, why do we need this optimization for size == 1?

Reply:
It is not needed. Maybe it is unrealistic that users have just one role? I thought for installations with many users most might just have one role and there it would help to avoid copying collections.

Reply:
Okay understand, thanks for the explanation. I think in most cases users will have more than one role, and we might be making the code a bit harder to read for a small performance benefit. Not a big deal of course.

Reply:
Reverted it now